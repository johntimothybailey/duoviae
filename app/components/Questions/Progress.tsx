import { QuestionsProps } from './props'
import { View } from 'react-native'
import { ProgressStyles } from './styles'
import { Text } from '@ui-kitten/components'
import { Bar as ProgressBar } from 'react-native-progress'
import React from 'react'

const Component = (props: QuestionsProps) => (
  <View style={ProgressStyles.container}>
    <Text style={ProgressStyles.text}>{props.step} / {props.list.length}</Text>
    <ProgressBar progress={(props.step ?? 0) / props.list.length} width={200} />
  </View>
)

export default Component
