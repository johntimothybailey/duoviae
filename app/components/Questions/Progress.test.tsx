import React from 'react'
import { cleanup, render } from '@testing-library/react-native'
import { ListOfQuestions } from '../../../test/fixtures/QuestionsStateMachine'
import Progress, { getPercentage } from './Progress'
import { BootComponent } from '../../../test'

describe('Questions > Progress Component', () => {
  describe('getPercentage()', () => {
    it('returns zero if length is zero', () => {
      expect(getPercentage(10, 0)).toEqual(0)
    })
    it('returns 6 as decimal', () => {
      expect(getPercentage(6, 10)).toEqual(0.6)
    })
  })
  describe('Render', () => {
    afterEach(cleanup)
    it('with an empty list', () => {
      const { queryByText, toJSON } = render(
        <BootComponent>
          <Progress list={[]} onSaveAnswer={() => {}} step={0} onNext={() => {}} />
        </BootComponent>
      )
      expect(queryByText('NaN')).toBeFalsy()
      expect(queryByText('0 / 0')).toBeTruthy()
      expect(toJSON()).toMatchSnapshot()
    })
    it('with a populated', () => {
      const { queryByText, toJSON } = render(
        <BootComponent>
          <Progress list={ListOfQuestions} onSaveAnswer={() => {}} step={0} onNext={() => {}} />
        </BootComponent>
      )
      expect(queryByText('NaN')).toBeFalsy()
      expect(queryByText('0 / 2')).toBeTruthy()
      expect(toJSON()).toMatchSnapshot()
    })
    it('with a populated starting at second question', () => {
      const { queryByText, toJSON } = render(
        <BootComponent>
          <Progress list={ListOfQuestions} onSaveAnswer={() => {}} step={1} onNext={() => {}} />
        </BootComponent>
      )
      expect(queryByText('NaN')).toBeFalsy()
      expect(queryByText('1 / 2')).toBeTruthy()
      expect(toJSON()).toMatchSnapshot()
    })
  })
})
