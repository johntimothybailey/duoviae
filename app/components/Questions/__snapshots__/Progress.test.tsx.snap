// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions > Progress Component Render with a populated 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E4E9F2",
          "borderRadius": 4,
          "borderWidth": 1,
          "flex": 0,
          "marginVertical": 24,
          "opacity": 1,
          "overflow": "hidden",
          "paddingBottom": 8,
          "paddingTop": 4,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": undefined,
            "height": 0,
          }
        }
      />
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flex": 0,
              "flexDirection": "row",
              "flexWrap": "nowrap",
              "justifyContent": "space-between",
              "paddingBottom": 16,
            }
          }
        >
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {},
              ]
            }
          >
            question
          </Text>
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {},
              ]
            }
          >
            0
             / 
            2
          </Text>
        </View>
        <View
          animated={true}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          indeterminate={false}
          indeterminateAnimationDuration={1000}
          onLayout={[Function]}
          progress={0}
          style={
            Array [
              Object {
                "backgroundColor": undefined,
                "borderColor": "#3366FF",
                "borderRadius": 4,
                "borderWidth": 1,
                "overflow": "hidden",
                "width": 200,
              },
              undefined,
            ]
          }
          useNativeDriver={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "#3366FF",
                "height": 6,
                "transform": Array [
                  Object {
                    "translateX": -7.105427357601002e-15,
                  },
                  Object {
                    "translateX": -99,
                  },
                  Object {
                    "scaleX": 0.0001,
                  },
                ],
              }
            }
          />
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Questions > Progress Component Render with a populated starting at second question 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E4E9F2",
          "borderRadius": 4,
          "borderWidth": 1,
          "flex": 0,
          "marginVertical": 24,
          "opacity": 1,
          "overflow": "hidden",
          "paddingBottom": 8,
          "paddingTop": 4,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": undefined,
            "height": 0,
          }
        }
      />
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flex": 0,
              "flexDirection": "row",
              "flexWrap": "nowrap",
              "justifyContent": "space-between",
              "paddingBottom": 16,
            }
          }
        >
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {},
              ]
            }
          >
            question
          </Text>
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {},
              ]
            }
          >
            1
             / 
            2
          </Text>
        </View>
        <View
          animated={true}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          indeterminate={false}
          indeterminateAnimationDuration={1000}
          onLayout={[Function]}
          progress={0.5}
          style={
            Array [
              Object {
                "backgroundColor": undefined,
                "borderColor": "#3366FF",
                "borderRadius": 4,
                "borderWidth": 1,
                "overflow": "hidden",
                "width": 200,
              },
              undefined,
            ]
          }
          useNativeDriver={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "#3366FF",
                "height": 6,
                "transform": Array [
                  Object {
                    "translateX": -7.105427357601002e-15,
                  },
                  Object {
                    "translateX": -49.5,
                  },
                  Object {
                    "scaleX": 0.50005,
                  },
                ],
              }
            }
          />
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Questions > Progress Component Render with an empty list 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E4E9F2",
          "borderRadius": 4,
          "borderWidth": 1,
          "flex": 0,
          "marginVertical": 24,
          "opacity": 1,
          "overflow": "hidden",
          "paddingBottom": 8,
          "paddingTop": 4,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": undefined,
            "height": 0,
          }
        }
      />
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 16,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flex": 0,
              "flexDirection": "row",
              "flexWrap": "nowrap",
              "justifyContent": "space-between",
              "paddingBottom": 16,
            }
          }
        >
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {},
              ]
            }
          >
            question
          </Text>
          <Text
            appearance="default"
            category="p1"
            style={
              Array [
                Object {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                Object {},
              ]
            }
          >
            0
             / 
            0
          </Text>
        </View>
        <View
          animated={true}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          indeterminate={false}
          indeterminateAnimationDuration={1000}
          onLayout={[Function]}
          progress={0}
          style={
            Array [
              Object {
                "backgroundColor": undefined,
                "borderColor": "#3366FF",
                "borderRadius": 4,
                "borderWidth": 1,
                "overflow": "hidden",
                "width": 200,
              },
              undefined,
            ]
          }
          useNativeDriver={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "#3366FF",
                "height": 6,
                "transform": Array [
                  Object {
                    "translateX": -7.105427357601002e-15,
                  },
                  Object {
                    "translateX": -99,
                  },
                  Object {
                    "scaleX": 0.0001,
                  },
                ],
              }
            }
          />
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
